### ISSUE TITLE
Cannot download the latest backup file for loading into the RDS instance

### ISSUE THREAD

**Issue by kencho51** [gigascience/gigadb-website#1902]
**Describe the bug**

**To Reproduce**
1. Follow the steps in `/docs/developers/SETUP_AWS_ENVIRONMENTS.md` to spin up staging servers
2. When `$ env OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ansible-playbook -i ../../inventories bastion_playbook.yml -e "gigadb_env=staging" -e "backupDate=latest"`, will get this error:
```
TASK [Load latest database into this environment's RDS instance] ******************************************************************************************************************************************************************
fatal: [3.36.204.163]: FAILED! => {"changed": true, "cmd": "./databaseReset.sh latest", "delta": "0:00:07.254492", "end": "2024-05-29 06:03:07.121971", "msg": "non-zero return code", "rc": 71, "start": "2024-05-29 06:02:59.867479", "stderr": "+ source .env\n++ PGDATABASE=gigadb\n++ PGHOST=rds-server-staging-ken.cjizsjwbxkxv.ap-northeast-2.rds.amazonaws.com\n++ PGPORT=5432\n++ PGUSER=gigadb\n++ PGPASSWORD='vagranT2021!'\n++ DB_PG_PASSWORD='vagranT2021!'\n++ DB_PG_USER=gigadb\n++ DB_PG_HOST=rds-server-staging-ken.cjizsjwbxkxv.ap-northeast-2.rds.amazonaws.com\n++ DB_PG_DATABASE=gigadb\n++ GIGADB_ENVIRONMENT=staging\n++ GITLAB_PROJECT_NAME=gigascience/forks/kencho-gigadb-website\n++ GITLAB_PROJECT_STRING=gigascience-forks-kencho-gigadb-website\n++ GITTER_IT_NOTIFICATION_ROOM_ID=63461e6b6da03739849db8e7\n++ GITTER_API_TOKEN=2bd3b2268c199f4e8c4ddc570012f97a2bce9ea8\n++ DEPLOYMENT_TIER=bastion\n++ date '--date=1 days ago' +%Y%m%d\n+ latest=20240528\n+ '[' -z x ']'\n+ '[' latest == latest ']'\n+ backupDate=20240528\n+ docker stop pg9_3\nError response from daemon: No such container: pg9_3\n+ true\n+ docker run --rm --detach --name pg9_3 -p 5432:5432 registry.gitlab.com/gigascience/forks/kencho-gigadb-website/production_pg9_3:staging\n+ docker run --rm --add-host=host.docker.internal:host-gateway -v /home/centos:/logs -v /home/centos/downloads:/downloads registry.gitlab.com/gigascience/forks/kencho-gigadb-website/production_updater:staging ./yii dataset-files/download-restore-backup --date 20240528 --force\nncftpget /gigadbv3_20240528.backup: server said: Failed to open file.", "stderr_lines": ["+ source .env", "++ PGDATABASE=gigadb", "++ PGHOST=rds-server-staging-ken.cjizsjwbxkxv.ap-northeast-2.rds.amazonaws.com", "++ PGPORT=5432", "++ PGUSER=gigadb", "++ PGPASSWORD='vagranT2021!'", "++ DB_PG_PASSWORD='vagranT2021!'", "++ DB_PG_USER=gigadb", "++ DB_PG_HOST=rds-server-staging-ken.cjizsjwbxkxv.ap-northeast-2.rds.amazonaws.com", "++ DB_PG_DATABASE=gigadb", "++ GIGADB_ENVIRONMENT=staging", "++ GITLAB_PROJECT_NAME=gigascience/forks/kencho-gigadb-website", "++ GITLAB_PROJECT_STRING=gigascience-forks-kencho-gigadb-website", "++ GITTER_IT_NOTIFICATION_ROOM_ID=63461e6b6da03739849db8e7", "++ GITTER_API_TOKEN=2bd3b2268c199f4e8c4ddc570012f97a2bce9ea8", "++ DEPLOYMENT_TIER=bastion", "++ date '--date=1 days ago' +%Y%m%d", "+ latest=20240528", "+ '[' -z x ']'", "+ '[' latest == latest ']'", "+ backupDate=20240528", "+ docker stop pg9_3", "Error response from daemon: No such container: pg9_3", "+ true", "+ docker run --rm --detach --name pg9_3 -p 5432:5432 registry.gitlab.com/gigascience/forks/kencho-gigadb-website/production_pg9_3:staging", "+ docker run --rm --add-host=host.docker.internal:host-gateway -v /home/centos:/logs -v /home/centos/downloads:/downloads registry.gitlab.com/gigascience/forks/kencho-gigadb-website/production_updater:staging ./yii dataset-files/download-restore-backup --date 20240528 --force", "ncftpget /gigadbv3_20240528.backup: server said: Failed to open file."], "stdout": "b3c28a5141cf71758a591867dc42adf001b5e3abc9a1abce04239952d677c4b1\n\nCommand is running for date 20240528\n\nDownloading production backup for 20240528\nFailed downloading backup file for date 20240528", "stdout_lines": ["b3c28a5141cf71758a591867dc42adf001b5e3abc9a1abce04239952d677c4b1", "", "Command is running for date 20240528", "", "Downloading production backup for 20240528", "Failed downloading backup file for date 20240528"]}

PLAY RECAP ************************************************************************************************************************************************************************************************************************
3.36.204.163               : ok=125  changed=39   unreachable=0    failed=1    skipped=10   rescued=0    ignored=0 
```

**Additional context**

The reason for this error is related to the https://github.com/gigascience/gigadb-website/issues/1893, the cronjob for `uploading database backup file to AWS FTP server` is commented out on 28/5/2024, so the latest backup created does not up upload to the `ftp://ec2-13-231-178-129.ap-northeast-1.compute.amazonaws.com`, as a result the latest available backup is `gigadbv3_20240527.backup`.

And in the `ftp_upload/upload.sh` script:
```
-bash-4.1$ sudo cat /home/peter/ftp_upload/upload.sh
#!/usr/bin/env bash

export PATH=/usr/local/bin/:$PATH
# Prepare params
latest=$(date +"%Y%m%d")
latest_backup_file="/var/lib/pgsql/database_dump/gigadbv3_$latest.backup"
credentials="user101:xxxxxxxxxxxx"
ftp_server="ftp://ec2-13-231-178-129.ap-northeast-1.compute.amazonaws.com"
node_monitor_ftp_server="ftp://ec2-54-249-152-250.ap-northeast-1.compute.amazonaws.com"

echo "Uploading $latest_backup_file file..."
curl -v -T $latest_backup_file -u $credentials $ftp_server
curl -v -T $latest_backup_file -u $credentials $node_monitor_ftp_server
```

The ftp server `ftp://ec2-13-231-178-129.ap-northeast-1.compute.amazonaws.com` is destroyed as seen in the [aws dashboard](https://ap-northeast-1.console.aws.amazon.com/ec2/home?region=ap-northeast-1#Instances:), so it leads to an error reported by ChrisH in an email  with subject `[tech-team] Fw: Returned mail: see transcript for details`
```
-bash-4.1$ sudo /bin/bash /home/peter/ftp_upload/upload.sh
Uploading /var/lib/pgsql/database_dump/gigadbv3_20240528.backup file...
* About to connect() to ec2-13-231-178-129.ap-northeast-1.compute.amazonaws.com port 21 (#0)
*   Trying 13.231.178.129... Connection timed out
* couldn't connect to host
* Closing connection #0

curl: (7) couldn't connect to host
* About to connect() to ec2-54-249-152-250.ap-northeast-1.compute.amazonaws.com port 21 (#0)
*   Trying 54.249.152.250... connected
* Connected to ec2-54-249-152-250.ap-northeast-1.compute.amazonaws.com (54.249.152.250) port 21 (#0)
< 220 Welcome to AWS FTP service.
> USER user101
< 331 Please specify the password.
> PASS gigaDB_!9274
< 230 Login successful.
> PWD
< 257 "/" is the current directory
* Entry path is '/'
> EPSV
* Connect data stream passively
< 229 Entering Extended Passive Mode (|||1029|)
*   Trying 54.249.152.250... connected
* Connecting to 54.249.152.250 (54.249.152.250) port 1029
> TYPE I
< 200 Switching to Binary mode.
> STOR gigadbv3_20240528.backup
< 150 Ok to send data.
} [data not shown]
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 85 40.7M    0     0   85 34.8M      0  7628k  0:00:05  0:00:04  0:00:01 8616k* Remembering we are in dir ""
< 226 Transfer complete.
100 40.7M    0     0  100 40.7M      0  7364k  0:00:05  0:00:05 --:--:-- 8399k* Connection #0 to host ec2-54-249-152-250.ap-northeast-1.compute.amazonaws.com left intact

> QUIT
< 221 Goodbye.
* Closing connection #0
```

So, this will work:
```
 % env OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ansible-playbook -i ../../inventories bastion_playbook.yml -e "gigadb_env=staging" -e "backupDate=20240527"
```

This is bug, as the old production gigadb website server will be decommissioned soon,  we should find a way to created the back up file and upload it to the ftp server.

**Comment by pli888** [comment]
Database backup files are being created by the live GigaDB environment and are now stored in the `gigadb-database-backups` S3 bucket. Therefore, we need to restore from a database backup file in this bucket when we are instantiating our personal GigaDB staging environments. This is what I have been doing when I need production data in my `dev` environment:
```
$ docker-compose run --rm test psql -h database -U gigadb -d gigadb -f "/var/www/gigadb/app/tools/files-metadata-console/sql/gigadb_gigascience-upstream-gigadb-website_live_20240527.backup"
```

> [!NOTE]
> Restoring this backup file takes a long time due to refreshing materialised view steps at the end of the database backup file.

**Comment by kencho51** [comment]
```
% cd gigadb-website
% ./up.sh
% cd gigadb/app/tools/files-url-updater
% docker-compose  run --rm updater ./vendor/bin/codecept run tests/functional
[+] Building 0.0s (0/0)                                                                                                                      docker:desktop-linux
[+] Building 0.0s (0/0)                                                                                                                      docker:desktop-linux
Codeception PHP Testing Framework v4.2.2 https://helpukrainewin.org
Powered by PHPUnit 8.5.31 by Sebastian Bergmann and contributors.

Warning: The use statement with non-compound name 'Yii' has no effect in /app/tests/functional/DatasetFilesCest.php on line 3

Functional Tests (10) --------------------------------------------------------------------------------------------------------------------------------------------
E DatasetFilesCest: Try download restore backup with date option 
E DatasetFilesCest: Try download no restore backup with date option 
E DatasetFilesCest: Try no download no restore backup with date option 
E DatasetFilesCest: Try no download restore backup with date option no local file 
E DatasetFilesCest: Try download restore backup with date option no remote file 
E DatasetFilesCest: Try download restore backup with latest option 
E DatasetFilesCest: Try no download restore backup with latest force options 
E DatasetFilesCest: Try restore command shows config 
E DatasetFilesCest: Try restore command with pending datasets abort 
E DatasetFilesCest: Try download restore default backup 
------------------------------------------------------------------------------------------------------------------------------------------------------------------
10x DEPRECATION: Calling the "Symfony\Component\BrowserKit\Client::getInternalResponse()" method before the "request()" one is deprecated since Symfony 4.1 and will throw an exception in 5.0. /app/vendor/symfony/browser-kit/Client.php:217


Time: 227 ms, Memory: 10.00 MB
```

**Comment by kencho51** [comment]
A quick fix has been implemented in PR #1866 by updating the `databaseReset.sh` to download the upstream backup file and load it into the RDS, eg:
```
docker run --rm --env-file .env -v /home/centos/.config/rclone/rclone.conf:/root/.config/rclone/rclone.conf -v /home/centos/restore:/restore --entrypoint /restore_database_from_s3_backup.sh registry.gitlab.com/$GITLAB_PROJECT_NAME/production_s3backup:$GIGADB_ENVIRONMENT "$backupDate"
```
